src/ServiceControl.Audit/Infrastructure/OWIN/Startup.cs

```diff
@@ -89,4 +91,40 @@ public override ICollection<Assembly> GetAssemblies()
 
         readonly Assembly additionalAssembly;
     }
+
+    /// <summary>
+    /// Replaces the <see cref="DefaultHttpControllerTypeResolver"/> with a similar implementation that allows non-public controllers.
+    /// </summary>
+    class InternalControllerTypeResolver : IHttpControllerTypeResolver
+    {
+        public ICollection<Type> GetControllerTypes(IAssembliesResolver assembliesResolver)
+        {
+            var controllerTypes = new List<Type>();
+            foreach (Assembly assembly in assembliesResolver.GetAssemblies())
+            {
+                if (assembly != null && !assembly.IsDynamic)
+                {
+                    Type[] source;
+                    try
+                    {
+                        source = assembly.GetTypes();
+
+                    }
+                    catch
+                    {
+                        continue;
+                    }
+
+                    controllerTypes.AddRange(source.Where(t => t != null && t.IsClass && !t.IsAbstract && typeof(IHttpController).IsAssignableFrom(t) && HasValidControllerName(t)));
+                }
+            }
+            return controllerTypes;
+        }
+
+        internal static bool HasValidControllerName(Type controllerType)
+        {
+            string controllerSuffix = DefaultHttpControllerSelector.ControllerSuffix;
+            return controllerType.Name.Length > controllerSuffix.Length && controllerType.Name.EndsWith(controllerSuffix, StringComparison.OrdinalIgnoreCase);
```

Comment:
We should probably have a unit test that looks for controllers that don't have this suffix because they'll just quietly not get picked up and it might be frustrating to figure out why.

Reply:
not opposed to that, but shouldn't we expect that every of the exposed APIs is somehow being used as part of acceptance tests anyway?